### DISCUSSION TITLE
How to best get class mapping for pickled StackedModel

### DISCUSSION THREAD

**Discussion by JoergPeisker** [biolab/orange3#5850] [Q&A] [answered]
Hi,

I am trying to deploy a stackedmodel classifier with a few dozen classes. The exporting and importing into a common python script works like a charm but how do I actually map the class-label back to the precited class-indices? Is there something like .classes_ or .class_indices?

**Comment by markotoplak** [comment, marked as answer]
To get predictions, the most Orange-like way is to prepare your data for prediction in an Orange Table and then just calling it with the data, for example as `stackedmodel(data)`. This returns class label indices, which you can then transform to values, like this:

```
import Orange
import Orange.classification
import Orange.ensembles.stack
iris = Orange.data.Table("iris")
lr = Orange.classification.LogisticRegressionLearner()
stack = Orange.ensembles.stack.StackedLearner([lr], aggregate=lr)
stackedmodel = stack(iris[:140])

# use the model
predictions = stackedmodel(iris[140:])
print(predictions)

# to get classes from indices
values = stackedmodel.domain.class_var.values
print([values[int(v)] for v in predictions])
```

Which returns

```
[2. 2. 2. 2. 2. 2. 2. 2. 2. 2.]
['Iris-virginica', 'Iris-virginica', 'Iris-virginica', 'Iris-virginica', 'Iris-virginica', 'Iris-virginica', 'Iris-virginica', 'Iris-virginica', 'Iris-virginica', 'Iris-virginica']
```

I hope I understood your question correctly.

**Reply by JoergPeisker** [reply]
Thanks for the fast reply and my apologize for the short question. This pretty much solves my problem :-)
